### PR TITLE
[feat](Nereids)  support refresh catalog command

### PR DESCRIPTION
**Pull request by vinlee19**
### What problem does this PR solve?

Issue Number: close # https://github.com/apache/doris/issues/42580

Related PR: #xxx

Problem Summary:
support refresh catalog command in Nereids
### Check List (For Author)

- Test 

    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by vinlee19** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
LGTM

**Review by LiBinfeng-01 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    public String toSql() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("REFRESH CATALOG ")
+                .append("`")
+                .append(catalogName)
+                .append("`");
+        return stringBuilder.toString();
+    }
+
+    public String getCatalogName() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java:129]: remove this method

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        return catalogName;
+    }
+
+    public Map<String, String> getProperties() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java:133]: remove this method

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    /**
+     * refresh catalog
+     */
+    public void handleRefreshCatalog() throws AnalysisException {
+        CatalogIf catalog = Env.getCurrentEnv().getCatalogMgr().getCatalogOrAnalysisException(catalogName);
+        CatalogLog log = new CatalogLog();
+        log.setCatalogId(catalog.getId());
+        log.setInvalidCache(isInvalidCache());
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java:94]:
```suggestion
        log.setInvalidCache(invalidCache);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    public String getCatalogName() {
+        return catalogName;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public boolean isInvalidCache() {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/refresh/RefreshCatalogCommand.java:137]: remove this method

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by starocean999** [comment]
please resolve the conflicts

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
@morningman PTAL

**Comment by LiBinfeng-01** [comment]
pls resolved conflicts

**Comment by vinlee19** [comment]
> pls resolved conflicts

done

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run buildall

**Comment by vinlee19** [comment]
run feut

**Comment by vinlee19** [comment]
run performance

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**